submodules/network/variables.tf

```diff
@@ -48,6 +53,25 @@ variable "cidr" {
   }
 }
 
+variable "pod_cidr" {
+  type        = string
+  default     = "100.64.0.0/16"
+  description = "The IPv4 CIDR block for the VPC."
+  validation {
+    condition = (
+      try(cidrhost(var.pod_cidr, 0), null) == regex("^(.*)/", var.pod_cidr)[0] &&
+      try(cidrnetmask(var.pod_cidr), null) == "255.255.0.0"
+    )
+    error_message = "Argument cidr must be a valid CIDR block."
+  }
+}
+
+variable "use_pod_cidr" {
+  type        = bool
+  description = "Use additional pod CIDR range (ie 100.64.0.0/16) for pod/service networking"
+  default     = true
+}
```

Comment:
More a philosophical question than anything, why two vars vs. an `optional(string)` pod_cidr where null means it's disabled?

Reply:
I actually started doing it that way, which looks something like this: https://github.com/dominodatalab/terraform-aws-eks/pull/22/files

However, I was a bit sketched out by what I had to do to the validation block:

```
    condition = (
      var.pod_cidr == null ||
      (try(cidrhost(var.pod_cidr, 0), null) == try(regex("^(.*)/", var.pod_cidr)[0], null) &&
      try(cidrnetmask(var.pod_cidr), null) == "255.255.0.0")
    )
```

Specifically, having to put the try on the right side as well. There were a couple minor things I didn't like (`pod_cidr` then works differently than `cidr`, additional exception to local variable calculation) that weren't a big deal, but that kind of made me decide not to go that route.

Reply:
I should also add that I kind of expect that the most likely user experience is toggling it on and off anyway, since there's no good reason to customize away from the carrier grade nat, so I'm sort of conceiving of the boolean as the standard user-facing variable and the other variable as the, uhh, "expert mode" variable.